lib/package-integrity/handler/integrity.ts

```diff
@@ -0,0 +1,287 @@
+      try {
+        execSync(`diff ${local} ${published}`, { stdio: ['ignore', 'inherit', 'inherit'] });
+      } catch (error) {
+        throw new Error(`${name} validation failed`);
+      }
+      console.log('Success');
+
+    } finally {
+      execSync(`rm -rf ${workdir}`);
+    }
+
+  }
+
+  private findOne(dir: string): string {
+    const files = fs.readdirSync(dir).filter(f => f.endsWith(this.ext));
+    if (files.length === 0) {
+      throw new Error(`No files found in ${dir}`);
+    }
+    if (files.length > 1) {
+      throw new Error(`Multiple files found in ${dir}`);
+    }
+    return path.join(dir, files[0]);
+  }
+
+}
+
+/**
+ * Properties for `RepositoryIntegrity`.
+ */
+export interface RepositoryIntegrityProps {
+  /**
+   * ARN of an AWS secrets manager secret containing a GitHub token.
+   */
+  readonly githubTokenSecretArn: string;
+
+  /**
+   * Repository slug (e.g cdk8s-team/cdk8s-core)
+   */
+  readonly repository: string;
+
+  /**
+   * Repository tag.
+   *
+   * @default - latest tag.
+   */
+  readonly tag?: string;
+
+  /**
+   * Prefix for detecting the latest tag of the repo. Only applies if `tag` isn't specified.
+   * This is useful for repositories that produce multiple packages, and hence multiple tags
+   * for example: https://github.com/cdk8s-team/cdk8s-plus/tags.
+   */
+  readonly tagPrefix?: string;
+
+  /**
+   * The projen task that produces the local artifacts.
+   *
+   * @default 'release'
+   */
+  readonly packTask?: string;
+}
+
+/**
+ * Integrity class for validating the artifacts produced by this repository against their published counterparts.
+ */
+export class RepositoryIntegrity {
+
+  constructor(private readonly props: RepositoryIntegrityProps) {}
+
+  /**
+   * Validate the artifacts of this repo against its published counterpart.
+   */
+  public validate() {
+
+    const repo = this.clone();
+    const artifacts = repo.pack(this.props.packTask);
+
+    let integrity = undefined;
+    for (const artifact of artifacts) {
+      switch (artifact.lang) {
+        case 'js':
+          integrity = new NpmArtifactIntegrity();
+          break;
+        case 'python':
+          integrity = new PyPIArtifactIntegrity();
+          break;
+        default:
+          break;
+      }
+      if (integrity) {
+        integrity.validate(artifact.directory);
+      }
+    }
+    console.log('Validation done');
+  }
+
+  private clone(): Repository {
+
+    const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'work'));
+    const token = execSync(`aws secretsmanager get-secret-value --secret-id ${this.props.githubTokenSecretArn} --output=text --query=SecretString`, { encoding: 'utf-8' }).toString().trim();
+    const repoDir = fs.mkdtempSync(path.join(workdir, 'repo'));
+
+    console.log(`Cloning ${this.props.repository} into ${repoDir}`);
+    execSync(`git clone https://${token}@github.com/${this.props.repository}.git ${repoDir}`);
```

Comment:
`token` should probably be optional. It's likely not needed for public repositories, unless your fleet gets throttled by GitHub on excessive activity.

Reply:
Done